pkg/kubelet/kubelet.go

```diff
@@ -2082,6 +2086,18 @@ func (kl *Kubelet) SyncTerminatedPod(ctx context.Context, pod *v1.Pod, podStatus
 	}
 	klog.V(4).InfoS("Pod termination unmounted volumes", "pod", klog.KObj(pod), "podUID", pod.UID)
 
+	if !kl.keepTerminatedPodVolumes {
+		if err := wait.PollUntilContextCancel(ctx, 100*time.Millisecond, true, func(ctx context.Context) (bool, error) {
```

Comment:
It might be worth a comment here that the volume manager relies on cleanup by consulting pod worker via `ShouldPodRuntimeBeRemoved` (https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go#L240). `ShouldPodRuntimeBeRemoved` should return true once `SyncTerminatingPod` returns, so it means that volume manager is expected to then cleanup the volumes in background.

Reply:
Agreed - comment saying what subsystem is expected to complete volume termination.

Reply:
Added a comment, let me know it this is enough.

Reply:
looks good, I would maybe just also mention that the volume manager specifically is responsible to perform this cleanup.